src/java.base/share/classes/java/lang/System.java

```diff
@@ -2334,77 +2052,31 @@ private static int initPhase2(boolean printToStderr, boolean printStackTrace) {
 
     /*
      * Invoked by VM.  Phase 3 is the final system initialization:
-     * 1. eagerly initialize bootstrap method factories that might interact
-     *    negatively with custom security managers and custom class loaders
```

Comment:
They might still interact negatively with custom class loaders though.

Reply:
The context here is custom a security manager doing string concatenation, this has required StringConcatFactory be eagerly initialized or security managers set on the command line to have been compiled with -XDstringConcat=inline. I think your question is about overriding the system class loader and whether its initialisation can use string concatenation reliably. That's a good thing to add a test for.